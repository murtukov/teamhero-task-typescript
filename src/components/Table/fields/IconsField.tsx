import React from 'react';
import { IFieldProps } from "./sharedProps";
import { styles as sharedStyles } from "./styles";
import { createUseStyles } from "react-jss";

export interface IIconsFieldProps extends IFieldProps {
    data?: any[],
    // The icon rendering method is defined by users
    renderer: (icon: any, index: number) => JSX.Element
}

function IconsField({data, renderer}: IIconsFieldProps) {
    const c = useStyles();

    if (undefined === data) {
        return null;
    }

    return (
        <td className={c.root}>
            <span className={c.wrapper}>
                {data.map((icon, i) => renderer(icon, i))}
            </span>
        </td>
    );
}

const useStyles = createUseStyles({
    ...sharedStyles,
    wrapper: {
        display: 'flex',
        justifyContent: 'space-between'
    }
})

export default IconsField;