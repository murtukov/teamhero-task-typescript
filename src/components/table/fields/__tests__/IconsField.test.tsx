import React from "react";
import { render } from "@testing-library/react";
import IconsField from "../IconsField";

const data = ['cross', 'plus', 'flash'];

function iconRenderer(name: string, i: number) {
    return <span key={i}>{name}</span>;
}

describe('IconsField component', () => {
    it('matches snapshot', () => {
        const { container } = render(<IconsField data={data} renderer={iconRenderer} source=''/>);
        expect(container.firstChild).toMatchSnapshot();
    });
})