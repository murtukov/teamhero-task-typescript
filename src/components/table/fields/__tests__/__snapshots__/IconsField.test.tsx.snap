// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconsField component renders correctly 1`] = `
<div
  class="cell-0-2-1 cell-d0-0-2-3"
>
  <span
    class="wrapper-0-2-2"
  >
    <span>
      cross
    </span>
    <span>
      plus
    </span>
    <span>
      flash
    </span>
  </span>
</div>
`;
